lib/components/app/app-menu.js

```diff
@@ -41,9 +47,92 @@ class AppMenu extends Component {
     window.location.href = startOverUrl
   }
 
+  _expandedPane = () => {
+    const { isPaneOpenLeft } = this.state
+    this.setState({ isPaneOpenLeft: !isPaneOpenLeft })
+  }
+
+  _expandSubmenu = () => {
+    const { isSubmenuExpanded } = this.state
+    this.setState({ isSubmenuExpanded: !isSubmenuExpanded })
+  }
+
+  _addExtraMenuItems = (menuItems) => {
+    const AddIconAndLabel = ({iconType, iconUrl, label}) => {
+      return (
+        iconUrl
+          ? <span>
+            <img alt={`icon for ${label} Menu Item`} src={iconUrl || 'bus'} />
+            {label}
+          </span>
+          : <span><Icon name={iconType || 'car'} />{label}</span>
+      )
+    }
+
+    return (
+      menuItems.map(menuItem => {
+        const {
+          children,
+          containsChildren,
+          href,
+          iconType,
+          iconUrl,
+          id,
+          label,
+          separator
+        } = menuItem
+        return (
+          containsChildren
+            ? <>
+              <MenuItem
+                className='expansion-button-container menu-item'
+                header
+                key={id}>
+                <button
+                  className='expand-submenu-button'
+                  onClick={this._expandSubmenu}>
+                  <AddIconAndLabel
+                    iconType={iconType}
+                    iconUrl={iconUrl}
+                    label={label} />
+                  <span>
+                    <Icon
+                      className='expand-menu-chevron'
+                      name={`chevron-${this.state.isSubmenuExpanded ? 'up' : 'down'}`} />
+                  </span>
+                </button>
+              </MenuItem>
+              <VelocityTransitionGroup
+                duration={50}
+                easing='linear'
+                enter={{ animation: 'slideDown' }}
+                leave={{ animation: 'slideUp' }}>
+                {this.state.isSubmenuExpanded && (
+                  <div className='sub-menu-container'>
+                    {this._addExtraMenuItems(children)}
+                  </div>
+                )}
+              </VelocityTransitionGroup>
+            </>
+            : <MenuItem
+              className={separator ? 'app-menu-separator menu-item' : 'menu-item'}
+              eventKey={href ? 1 : ''}
```

Comment:
What does this do? If we keep it, it can be shortened like this

```suggestion
              eventKey={href && 1}
```

Reply:
Hmm I dont think it is needed as we are not using an active state. Will remove.